### PR TITLE
 NB: Simplify group_download_as implementation and make it bigtoolitem

### PR DESCRIPTION
**Pull request by pedropintosilva**
Revamped https://github.com/CollaboraOnline/online/pull/4576

commit 2d3313813488d23b7fd675e7ac08f2dfb07713bb (HEAD -> private/pedro/group_download_as-v2, origin/private/pedro/group_download_as-v2)
Author: Pedro Pinto Silva <pedro.silva@collabora.com>
Date:   Fri Apr 8 18:19:37 2022 +0200

     NB: Simplify group_download_as implementation and make it bigtoolitem
    
    - Avoid using random UNO command (InsertGraphic) for download action
    - Centralize download option (re-use bigmenubartoolitem)
    
    Signed-off-by: Pedro Pinto Silva <pedro.silva@collabora.com>
    Change-Id: Ie9cb8564acee8611221e0ff6f841de7b31b777e4

### CODE REVIEW

File: browser/src/control/Control.NotebookbarCalc.js [hunk 1]
```diff
@@ -152,18 +152,10 @@ L.Control.NotebookbarCalc = L.Control.NotebookbarWriter.extend({
...
-						'id': 'downloadas',
-						'type': 'menubartoolitem',
-						'text': _('Download'),
-						'command': '.uno:InsertGraphic'
-					}
-				]
+				'id': 'downloadas',
+				'type': 'bigmenubartoolitem',
+				'text': _('Download'),
+				'enabled': 'true'
```
Comment (pedropintosilva) [inline browser/src/control/Control.NotebookbarCalc.js:158]: I don't think we need to expressly enable this

File: browser/src/control/Control.NotebookbarImpress.js [hunk 2]
```diff
@@ -217,18 +217,10 @@ L.Control.NotebookbarImpress = L.Control.NotebookbarWriter.extend({
...
-						'id': 'downloadas',
-						'type': 'menubartoolitem',
-						'text': _('Download'),
-						'command': '.uno:InsertGraphic'
-					}
-				]
+				'id': 'downloadas',
+				'type': 'bigmenubartoolitem',
+				'text': _('Download'),
+				'enabled': 'true'
```
Comment (pedropintosilva) [inline browser/src/control/Control.NotebookbarImpress.js:223]: I don't think we need to expressly enable this

File: browser/src/control/Control.NotebookbarWriter.js [hunk 3]
```diff
@@ -165,18 +165,10 @@ L.Control.NotebookbarWriter = L.Control.Notebookbar.extend({
...
-						'id': 'downloadas',
-						'type': 'menubartoolitem',
-						'text': _('Download'),
-						'command': '.uno:InsertGraphic'
-					}
-				]
+				'id': 'downloadas',
+				'type': 'bigmenubartoolitem',
+				'text': _('Download'),
+				'enabled': 'true'
```
Comment (pedropintosilva) [inline browser/src/control/Control.NotebookbarWriter.js:171]: I don't think we need to expressly enable this

File: browser/src/control/Control.NotebookbarImpress.js (hunk 2 above)
Comment (pedropintosilva) [inline browser/src/control/Control.NotebookbarImpress.js:223]: I don't think we need to expressly enable this

File: browser/src/control/Control.NotebookbarCalc.js (hunk 1 above)
Comment (pedropintosilva) [inline browser/src/control/Control.NotebookbarCalc.js:158]: I don't think we need to expressly enable this

**Comment by pedropintosilva** [comment]
Also removed `'enabled': 'true'` previously added line from each `downloadas` block in L.Control.NotebookbarWriter/Calc/Impress.js

File: browser/src/control/Control.JSDialogBuilder.js [hunk 4]
```diff
@@ -2213,7 +2213,6 @@ L.Control.JSDialogBuilder = L.Control.extend({
 			$(control.container).unbind('click');
 			$(control.container).click(function () {
 				$(control.container).w2menu({
-					items: builder._menus[menuId],
```
Comment (eszkadev) [inline browser/src/control/Control.JSDialogBuilder.js:2216]: This should stay here to show menu for menubartoolitem.
I assume this is added by mistake.

File: browser/src/control/Control.NotebookbarCalc.js [hunk 5]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (eszkadev) [inline browser/src/control/Control.NotebookbarCalc.js:1]: this empty line shoudn't be here

File: browser/src/control/Control.JSDialogBuilder.js (hunk 4 above)
Comment (pedropintosilva) [reply browser/src/control/Control.JSDialogBuilder.js]: yup, my mistake

**Review by eszkadev — APPROVED** [review]
looks good

**Review by lpranam — APPROVED** [review]

**Pull request merged**